extensions/artifacts/org.wso2.developerstudio.codenvy.ext.appfactory/src/main/java/org/wso2/developerstudio/codenvy/ext/appfactory/server/client/HTTPAppFactoryClient.java

```diff
@@ -0,0 +1,224 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.wso2.developerstudio.codenvy.ext.appfactory.server.client;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.conn.ClientConnectionManager;
+import org.apache.http.conn.scheme.Scheme;
+import org.apache.http.conn.scheme.SchemeRegistry;
+import org.apache.http.conn.ssl.SSLSocketFactory;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.impl.conn.tsccm.ThreadSafeClientConnManager;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.developerstudio.codenvy.ext.appfactory.server.exception.AppFactoryServerException;
+import org.wso2.developerstudio.codenvy.ext.appfactory.shared.dto.AppFactoryLoginResponse;
+import org.wso2.developerstudio.codenvy.ext.appfactory.shared.dto.ErrorType;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Handles all type of HTTP/HTTPS requests and responses from DevStudio to AppFactory/ AppCloud server
+ */
+public class HTTPAppFactoryClient {
+    private static final Logger log = LoggerFactory.getLogger(HTTPAppFactoryClient.class);
+
+    private static HttpClient client;
+    private static String cookie;
+
+    /**
+     * Creates an HTTP Entity using given request parameters
+     *
+     * @param serverURL          AppFactory/ AppCloud server URL to connect with for login request
+     * @param loginRequestParams Map that contains login request params as key value pairs
+     * @return Populated AppFactoryLoginResponse object
+     * @throws org.wso2.developerstudio.codenvy.ext.appfactory.server.exception.AppFactoryServerException
+     */
+    public static AppFactoryLoginResponse doLogin(String serverURL, Map<String, String> loginRequestParams)
+            throws AppFactoryServerException {
+        AppFactoryLoginResponse loginResponse = new AppFactoryLoginResponse();
+        String successLogin;
+
+        //Creating an http client and wrap it
+        client = new DefaultHttpClient();
```

Comment:
Again, why is a HttpClient object is created over and over again when a log-in request comes in? Also, updating a globally referenced variable this way is always troublesome. A better approach would be to initialize an instance of HttpClient within the constructor or some other initialization logic and then re-use it in everywhere else in the code. Maybe it's a good idea to make certain parameters associated with HttpClient such as maximum connection counts, etc configurable through some configuration file as well.